src/components/Api.js

```diff
@@ -0,0 +1,115 @@
+import { data } from "autoprefixer";
+
+class Api {
+  constructor(initialCards) {
+    this._initialCards = initialCards;
+  }
+
+  handleServerResponse(response) {
+    if (response.ok) {
+      return response.json();
+    }
+    return Promise.reject(`Error: ${response.status}`);
+  }
+
+  getInitialCards() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/cards", {
+      headers: {
+        authorization: "07909f6e-76be-4aa7-8439-3e97a34a8c13",
+      },
+    }).then(this.handleServerResponse);
+  }
+
+  userInfoReq() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/users/me", {
+      method: "GET",
+      headers: {
+        authorization: "07909f6e-76be-4aa7-8439-3e97a34a8c13",
+      },
+    }).then(this.handleServerResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this.handleServerResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.